src/rust/bitbox02-rust/src/apps/bitcoin/keypath.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::pb;
+pub use pb::btc_script_config::multisig::ScriptType as MultisigScriptType;
+
+use util::bip32::HARDENED;
+
+const BIP44_ACCOUNT_MIN: u32 = HARDENED;
+const BIP44_ACCOUNT_MAX: u32 = HARDENED + 99; // 100 accounts
+
+const PURPOSE_MULTISIG: u32 = 48 + HARDENED;
+const MULTISIG_SCRIPT_TYPE_P2WSH: u32 = 2 + HARDENED;
+const MULTISIG_SCRIPT_TYPE_P2WSH_P2SH: u32 = 1 + HARDENED;
+
+/// Validates a keypath to be
+/// m/expected_purpose/expected_coin/account, where account between 0' and 99'.
+pub fn validate_account(
+    keypath: &[u32],
+    expected_coin: u32,
+    expected_purpose: u32,
+) -> Result<(), ()> {
+    if let [purpose, coin, account] = *keypath {
+        if purpose == expected_purpose
+            && coin == expected_coin
+            && account >= BIP44_ACCOUNT_MIN
+            && account <= BIP44_ACCOUNT_MAX
+        {
+            return Ok(());
+        }
+    }
+    Err(())
+}
+
+/// Validates a multisig keypath.
+/// Supported:
+/// - Electrum-style: m/48'/coin'/account'/script_type', where script_type is 1 for p2wsh-p2sh and 2
+///   for p2wsh.
+pub fn validate_account_multisig(
+    keypath: &[u32],
+    expected_coin: u32,
+    script_type: MultisigScriptType,
+) -> Result<(), ()> {
+    if keypath.len() == 4 {
```

Comment:
not sure about idiomatic rust but I personally find code with "minimal indentation" easier to read:

```rust
if keypath.len() != 4 {
  return Err(())
}
validate_account(...)?;
...
```

less jumping up and down to see which "if" block is which.
and the same in `validate_address_multisig`.

Reply:
i will extend this in the next pr with a `if keypath.len() == 3` section that matches `m/48'/coin'/accoint'` keypaths for Nunchuk.